### PR TITLE
[API] add original price to product variant

### PR DESCRIPTION
**Pull request by SirDomin**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Shop/ProductContext.php [hunk 1]
```diff
@@ -146,6 +146,27 @@ public function iShouldSeeTheProductPrice(int $price): void
...
+                [$this->responseChecker->getResponseContent($this->client->getLastResponse())],
+                $originalPrice
+            )
+        );
+    }
+
+    /**
+     * @Then I should not see any original price
+     */
+    public function iShouldNotSeeTheProductOriginalPrice(): void
```
Comment (arti0090) [inline src/Sylius/Behat/Context/Api/Shop/ProductContext.php:165]:
```suggestion
    public function iShouldNotSeeAnyOriginalPrice(): void
```

**Review by arti0090 — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Serializer/ProductVariantNormalizer.php [hunk 2]
```diff
@@ -71,6 +71,12 @@ public function normalize($object, $format = null, array $context = [])
             unset($data['price']);
         }
 
+        $originalPrice = $this->priceCalculator->calculateOriginal($object, ['channel' => $this->channelContext->getChannel()]);
+
+        if ($originalPrice !== $data['price']) {
+            $data['original_price'] = $originalPrice;
+        }
```
[the remark this answers is not in the document]
Comment (AdamKasp) [reply src/Sylius/Bundle/ApiBundle/Serializer/ProductVariantNormalizer.php:78]: And do we really want to do it in this way? IMO JSON should always look the same way, if `originalPrice` is configured it should show this value if not it should show maybe `null`. From a frontend perspective JSON with a "random" body will cause unexpected errors (I always check the value of the field instead the existence of this field)

**Review by AdamKasp — CHANGES_REQUESTED** [review]

File: src/Sylius/Behat/Context/Api/Shop/ProductVariantContext.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Webmozart\Assert\Assert;
+
+final class ProductVariantContext implements Context
+{
+    /** @var ApiClientInterface */
+    private $client;
```
Comment (Arminek) [inline src/Sylius/Behat/Context/Api/Shop/ProductVariantContext.php:25]: Can we start using typed properties?

Comment (SirDomin) [reply src/Sylius/Behat/Context/Api/Shop/ProductVariantContext.php:25]: sounds good to me

File: src/Sylius/Bundle/ApiBundle/Serializer/ProductVariantNormalizer.php [hunk 4]
```diff
@@ -67,8 +67,13 @@ public function normalize($object, $format = null, array $context = [])
 
         try {
             $data['price'] = $this->priceCalculator->calculate($object, ['channel' => $this->channelContext->getChannel()]);
+            $originalPrice = $this->priceCalculator->calculateOriginal($object, ['channel' => $this->channelContext->getChannel()]);
+
+            $data['original_price'] = ($data['price'] === $originalPrice ? null : $originalPrice);
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Serializer/ProductVariantNormalizer.php:72]: but if `originalPrice` is equal to `price` we should show 2 same value for `price` and `originalPrice` with this code, you will overwrite `originalPrice` :(

what if you will only leave ` data['original_price'] = $originalPrice`

**Review by AdamKasp — CHANGES_REQUESTED** [review]

**Review by AdamKasp — APPROVED** [review]

**Review by Tomanhez — APPROVED** [review]

**Comment by AdamKasp** [comment]
Thanks, @SirDomin! :1st_place_medal:

File: src/Sylius/Behat/Context/Api/Shop/ProductContext.php [hunk 5]
```diff
@@ -343,6 +366,17 @@ private function hasProductWithName(array $products, string $name): bool
         return false;
     }
 
+    private function hasProductWithOriginalPrice(array $products, int $originalPrice): bool
+    {
+        foreach ($products as $product) {
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Shop/ProductContext.php:371]: Why do we have an array of products here? Also, this loop just checks if there is any product with `original_price`, not the particular, which seems bad

**Pull request merged**